rsfbclient-rust/src/blr.rs

```diff
@@ -59,17 +59,19 @@ pub fn params_to_blr(
     for p in params {
         match p {
             Param::Text(s) => {
-                if s.len() > MAX_DATA_LENGTH {
+                let len = s.len();
+                let bytes = conn.charset.encode(s.to_string())?;
```

Comment:
The length is wrong, you need to use the length after encoding, not before.
This is what was causing the code to hang, as it was sending a malformed message

```suggestion
                let bytes = conn.charset.encode(s.to_string())?;
                let len = bytes.len();
```

Reply:
Fixed!